main_sweeps.py

```diff
@@ -0,0 +1,233 @@
+                        os.makedirs(output_sensor_path)
+                    sensor_data = nusc.get("sample_data", sensor_token)
+                    file_path = os.path.join(NUSCENES_ROOT, sensor_data["filename"])
+                    i = 0
+                    if sensor == "LIDAR_TOP":
+                        # nuscenes lidar data is stored as (x, y, z, intensity, ring index)
+                        while i < nsweeps_lidar and sensor_token != "":
+                            sensor_data = nusc.get("sample_data", sensor_token)
+                            file_path = os.path.join(
+                                NUSCENES_ROOT, sensor_data["filename"]
+                            )
+                            timestamp = round_to_micros(sensor_data["timestamp"])
+                            # Not always exactly 10
+                            if (
+                                sensor_data["is_key_frame"] and i != 0
+                            ) or sample_token == "":
+                                break
+                            scan = np.fromfile(file_path, dtype=np.float32)
+                            points = scan.reshape((-1, 5))
+
+                            # Transform lidar points from point sensor frame to egovehicle frame
+                            calibration = nusc.get(
+                                "calibrated_sensor",
+                                sensor_data["calibrated_sensor_token"],
+                            )
+                            egovehicle_R_lidar = quat2rotmat(calibration["rotation"])
+                            egovehicle_t_lidar = np.array(calibration["translation"])
+                            egovehicle_SE3_lidar = SE3(
+                                rotation=egovehicle_R_lidar,
+                                translation=egovehicle_t_lidar,
+                            )
+                            points_egovehicle = (
+                                egovehicle_SE3_lidar.transform_point_cloud(
+                                    points[:, :3]
+                                )
+                            )
+
+                            write_ply(
+                                points_egovehicle, points, output_sensor_path, timestamp
+                            )
+
+                            if not os.path.isfile(
+                                os.path.join(
+                                    poses_path, f"city_SE3_egovehicle_{timestamp}.json"
+                                )
+                            ):
+                                ego_pose = nusc.get(
+                                    "ego_pose", sensor_data["ego_pose_token"]
+                                )
+                                ego_pose_dict = {
+                                    "rotation": ego_pose["rotation"],
+                                    "translation": ego_pose["translation"],
+                                }
+                                with open(
```

Comment:
i think it's a bit cleaner to abstract this part away
```python
from argoverse.utils.json_utils.py import save_json_dict
save_json_dict( os.path.join(poses_path, f"city_SE3_egovehicle_{timestamp}.json"), ego_pose_dict)
```

Reply:
to avoid all the json.dump lines

Reply:
Done